heartbeat/monitors/active/icmp/icmp.go

```diff
@@ -82,25 +107,25 @@ func create(
 			return nil, 0, err
 		}
 
-		jobs = append(jobs, wrappers.WithURLField(u, job))
+		j = append(j, wrappers.WithURLField(u, job))
 	}
 
-	return jobs, len(config.Hosts), nil
+	return j, len(jf.config.Hosts), nil
 }
 
-func createPingIPFactory(config *Config) func(*beat.Event, *net.IPAddr) error {
-	return func(event *beat.Event, ip *net.IPAddr) error {
-		rtt, n, err := loop.ping(ip, config.Timeout, config.Wait)
+func (jf *jobFactory) pingIPFactory(config *Config) func(*net.IPAddr) jobs.Job {
+	return monitors.MakePingIPFactory(func(event *beat.Event, ip *net.IPAddr) error {
+		rtt, n, err := jf.loop.ping(ip, config.Timeout, config.Wait)
 		if err != nil {
 			return err
 		}
 
 		icmpFields := common.MapStr{"requests": n}
 		if err == nil {
 			icmpFields["rtt"] = look.RTT(rtt)
-			eventext.MergeEventFields(event, icmpFields)
```

Comment:
This was the main issue in the original code, we merged `icmpFields` into the root of the event, when it should have been nested in the `"icmp"` key.